pkg/bgpv1/types/bgp.go

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package types
+
+import (
+	"context"
+	"net"
+
+	"github.com/cilium/cilium/api/v1/models"
+	v2alpha1api "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+)
+
+// BGPGlobal contains high level BGP configuration for given instance.
+type BGPGlobal struct {
+	ASN                   uint32
+	RouterID              string
+	ListenPort            int32
```

Comment:
Better to document why this is signed 32bit (port number is unsigned 16bit in principle) and `-1` has a special meaning.